azkaban-common/src/main/java/azkaban/executor/container/watch/KubernetesWatch.java

```diff
@@ -186,13 +187,21 @@ public void initializeAndStartPodWatch() {
    *
    * @return
    */
-  public Thread launchPodWatch() {
+  public synchronized Thread launchPodWatch() {
```

Comment:
Can you please add a comment that the caller must handle the exception otherwise it could cause unexpected failures.

Reply:
Done.